### PR TITLE
[docs] Fix release notes build failures

### PR DESCRIPTION
**Pull request by darrenchan**
Move _deconst.json and deconst-requirements.txt to
source directory to fix build failures in Strider. This
will enable release note build preview in PRs and a draft preview at
https://staging.developer.rackspace.com/staging.horse/review/docs/private-cloud/rpc/release-notes/

### CODE REVIEW

**Comment by darrenchan** [comment]
Addresses https://github.com/rackerlabs/nexus-control/issues/555.

CCing @smashwilson @meker12

File: releasenotes/source/_deconst.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-    "contentIDBase": "https://github.com/rcbops/rpc-openstack/v13",
+    "contentIDBase": "https://github.com/rcbops/rpc-openstack/v13”,
```
Comment (smashwilson) [inline releasenotes/source/_deconst.json]: Smart quotes strike again!

**Comment by hughsaunders** [comment]
recheck_all

File: releasenotes/source/_deconst.json (hunk 1 above)
Comment (darrenchan) [inline releasenotes/source/_deconst.json]: Thanks, I thought I disabled this on my mac!

**Comment by hughsaunders** [comment]
👍

**Comment by darrenchan** [comment]
@smashwilson  sorry, Strider is still showing a build error:

> Deconst

 content directory: /workspace/data/rcbops-rpc-openstack-master/job-57be323f3ddba90100030888/releasenotes/source
Unable to parse _deconst.json file in /workspace/data/rcbops-rpc-openstack-master/job-57be323f3ddba90100030888/releasenotes/source/_deconst.json
Oh no: [Error: Unable to parse _deconst.json]

>

**Comment by rackernexus** [comment]
Your build succeeded, but some of your content is not mapped anywhere in the control repository yet! See the [Deconst documentation](https://deconst.horse/writing-docs/coordinator/mapping/) for information about content mapping.

**Comment by rackernexus** [comment]
Your build succeeded, but some of your content is not mapped anywhere in the control repository yet! See the [Deconst documentation](https://deconst.horse/writing-docs/coordinator/mapping/) for information about content mapping.

**Comment by rackernexus** [comment]
Your build succeeded, but some of your content is not mapped anywhere in the control repository yet! See the [Deconst documentation](https://deconst.horse/writing-docs/coordinator/mapping/) for information about content mapping.

**Comment by meker12** [comment]
@darrenchan   So 


template specified in conf.py must match template specified in routes.d

Currently, your conf.py has:

# The builder to use when running via the deconst preparer
# builder = 'deconst-single'
builder = 'deconst-serial'

routes.d for the development server specifies:   

"^/docs/private-cloud/release-notes/": "developer.rackspace.com/docs-singlepage.html",



contentIDbase specified in _deconst.json must match the ID specified in the config.json file

Also, this project is currently defined to build on the development server staging.horse.  
This enables the PR preview, but it doesn't enable publishing to the developer.rackspace.com.  Is that your intent?

If you want to deploy to production, then you need to update the routes.d and config.d files for developer.rackspace.com site. 

@smashwilson  Can a repository be configured for both the staging.horse site AND the developer.rackspace.com site?  Or do you need to delete the staging.horse configuration, and then update the developer.rackspace.com configuration.

**Comment by darrenchan** [comment]
@meker12  Thanks for the explanation and your patience.  The aim is for:
- master branch to have PR preview and a draft preview like [docs-rpc ](https://staging.developer.rackspace.com/staging.horse/review/docs/private-cloud/rpc/master/)
- mitaka-13.0 branch to have PR preview and published to developer.rackspace.com.

@robb-romans @erikmwilson @klevenstein feel free to add your thoughts.

**Comment by rackernexus** [comment]
Your build succeeded, but some of your content is not mapped anywhere in the control repository yet! See the [Deconst documentation](https://deconst.horse/writing-docs/coordinator/mapping/) for information about content mapping.

**Comment by meker12** [comment]
@darrenchan @darrenchan Based on experience fixing the [RPC docs internal preview](https://github.com/rackerlabs/docs-rpc/pull/529) , you might want to do a fresh PR for the Release Notes with the changes you added in this one. 

Also, to add a production preview, you have to add the mitaka branch to the [content.repositories.json](https://github.com/rackerlabs/nexus-control/edit/master/content-repositories.json) configuration.  Then, update the config.d and routes.d files for the developer.rackspace.com site.  

It probably makes sense to get the development preview working seamlessly before configuring the production one.  Also, the steps to configure these are documented in the [Contributor Guide](http://rackerlabs.github.io/docs-rackspace/tools/publishing.html#update-nexus-control-configuration). Let us know if that doc isn't clear, and either update it or submit an issue with your suggestions for improvement.   

Hope this helps!

**Comment by darrenchan** [comment]
Thanks @meker12. I been using the [Contributor Guide](http://rackerlabs.github.io/docs-rackspace/tools/publishing.html#update-nexus-control-configuration)  and deconst.horse to understand the process.

**Comment by meker12** [comment]
I think the content is probably a bit confusing, so if you have suggestions for making it better, please do.   Also, I created an issue: https://github.com/rackerlabs/nexus-control/issues/566
for the Mitaka update.  If that is clear enough, maybe we can add it to the contribu guide.

**Comment by smashwilson** [comment]
> @smashwilson Can the same branch of a repository be configured for both the staging.horse site AND the developer.rackspace.com site? Or do you need to delete the staging.horse configuration, and then update the developer.rackspace.com configuration.

You can indeed! You can map the same content ID to as many domains and paths as you want. The PR builder should give you all links for the content on each build.

**Pull request closed without merging**